### PR TITLE
planner: fix oom by LRUPlanCache's key expand

### LINKED ISSUE #38244: tidb oom when add and drop column repeatly
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
add and drop column repeatly

### 2. What did you expect to see? (Required)
not crash
test plan：https://tcms.pingcap.net/dashboard/executions/plan/1291915
https://clinic.pingcap.com.cn/grafana/d/000000011cloud/clinic-operator-cluster-tidb?orgId=1&var-tidb_cluster=7148262878439930296&var-instance=All&var-store=All&from=1664334765659&to=1664353989204&var-tidb_version=master&var-org=31

### 3. What did you see instead (Required)
tidb oom

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v6.3.0-alpha
Edition: Community
Git Commit Hash: c5a28bfea84f727662f339d7969ff6f0a68f7cfe
Git Branch: heads/refs/tags/v6.3.0-alpha
UTC Build Time: 2022-09-27 11:08:51
GoVersion: go1.19.1
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: close #38244 

Problem Summary:
The plan cache key change with `schemaversion` change, when evict the expired plan I did't delete the expired key causing the memory expand
### What is changed and how it works?
Remove the expired key in LRU
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```
### Manual test (detailed scripts )
* Set plancache with default size(100).
* Use sysbench do oltp_write_only.
* Use a Python script add/drop column in table `sbtest1` all the time, see the bucket size in debug mode.
```
# sysbench command 
sysbench --mysql-user=root --db-driver=mysql --time=0 --report-interval=10 --mysql-db=sysbench --mysql-host=127.0.0.1  --mysql-port=4000 --threads=8 --tables=4 --table-size=10000 --mysql-ignore-errors=2013,1213,1105,1205,8022,8027,8028,9004,9007 --rand-type=special oltp_write_only 
```
Before:
飞书20220930-150304
After:
飞书20220930-150307

### CODE REVIEW

**Review by qw4990 — COMMENTED** [review]
Please add an UT for this case.

**Review by lcwangchao — APPROVED** [review]

File: planner/core/plan_cache_lru.go [hunk 1]
```diff
@@ -183,8 +183,12 @@ func (l *LRUPlanCache) removeOldest() {
 
 // removeFromBucket remove element from bucket
 func (l *LRUPlanCache) removeFromBucket(element *list.Element) {
-	bucket := l.buckets[string(hack.String(element.Value.(*planCacheEntry).PlanKey.Hash()))]
+	hash := string(hack.String(element.Value.(*planCacheEntry).PlanKey.Hash()))
```
Comment (qw4990) [inline planner/core/plan_cache_lru.go:186]: How about creating a dedicated function like `strHashKey(k kvcache.Key, deepCopy bool)` to do this?

**Review by qw4990 — APPROVED** [review]

**Pull request merged**